modules/core/include/opencv2/core/types.hpp

```diff
@@ -63,6 +63,16 @@ namespace cv
 //! @addtogroup core_basic
 //! @{
 
+//////////////////////////////// Enum operators //////////////////////////////
+
+template<typename T, typename std::enable_if< std::is_enum<T>::value >::type* = nullptr>
```

Comment:
Maybe consider adding this only for "flags" enums on per-enum basis. For "mode" enums this does not make sense. It could confuse users to use this to pass OR-ed values where it should not be used.

Reply:
Also `operator | ()` is needed too.